talk/morelanguage/move.tex

```diff
@@ -156,16 +158,16 @@
     \end{cppcode*}
   \end{block}
   \pause
-  \begin{block}{There are mainly two ways}
+  \begin{block}{Explicitly request moving}
     \begin{itemize}
-    \item casting to an rvalue reference
-    \item using the std::move function
+    \item using the \mintinline{cpp}{std::move} function
+    \item which is basically a cast to an rvalue reference
     \end{itemize}
     \begin{cppcode*}{firstnumber=6}
-      T a;
-      T b = a;                   // Copy assign
-      T c = static_cast<T&&>(a); // Move assign
-      T d = std::move(a);        // Move assign
+      T a, b;
+      b = a;                   // Copy assign
+      b = std::move(a);        // Move assign
+      b = static_cast<T&&>(a); // Move assign (don't)
```

Comment:
Funnily enough, `T b = a` is a copy construction, because the compiler rewrites this line to `T b(a)`;

Reply:
That's correct. Maybe change it back and label as `Copy construction`?

We don't need to mention that in the lecture, but if somebody asks, I would be happy to explain.

Reply:
I changed the snippet now to use the `swap` implementation on the same slide and labelled what operations are performed accordingly.